### ISSUE TITLE
Can not migrate Database.

### ISSUE THREAD

**Issue by BlueSpider67** [inaturalist/inaturalist#1256]
https://www.dropbox.com/s/77m3oqrq5z1hrgl/Screenshot%202017-01-11%2012.17.26.png?dl=0

Please check this screenshot

Thanks

**Comment by kueda** [comment]
Are you initializing a new database or trying to run modifications on an existing one? If the former, use rake `db:setup` instead. That will load the schema in from structure.sql instead of trying to run all the migrations. If the latter, I'd just try to fake it out by creating the flags table.

**Comment by BlueSpider67** [comment]
Thanks, Let me try again

**Comment by BlueSpider67** [comment]
Could you provide me guide on the current version?

**Comment by kueda** [comment]
I don't know what that means. Try running `rake db:setup` and if that doesn't work please paste in any error messages you're seeing.

**Comment by BlueSpider67** [comment]
PG::InvalidCatalogName: ERROR:  template database "template_postgis" does not exist
: CREATE DATABASE "inaturalist_development" ENCODING = 'utf8' TEMPLATE = "template_postgis"
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql/database_statements.rb:155:in `async_exec'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql/database_statements.rb:155:in `block in execute'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract_adapter.rb:472:in `block in log'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-4.2.6/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract_adapter.rb:466:in `log'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql/database_statements.rb:154:in `execute'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql/schema_statements.rb:78:in `create_database'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/bundler/gems/activerecord-postgis-adapter-4d7c4c91720e/lib/active_record/connection_adapters/postgis_adapter/postgis_database_tasks.rb:24:in `create'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:93:in `create'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:107:in `block in create_current'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:275:in `block in each_current_configuration'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:274:in `each'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:274:in `each_current_configuration'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:106:in `create_current'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/railties/databases.rake:17:in `block (2 levels) in <top (required)>'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:240:in `block in execute'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:235:in `each'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:235:in `execute'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:172:in `invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:201:in `block in invoke_prerequisites'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:199:in `each'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:199:in `invoke_prerequisites'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:178:in `block in invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:172:in `invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:201:in `block in invoke_prerequisites'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:199:in `each'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:199:in `invoke_prerequisites'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:178:in `block in invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:172:in `invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:165:in `invoke'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:150:in `invoke_task'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:106:in `each'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:106:in `block in top_level'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:115:in `run_with_threads'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:100:in `top_level'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:78:in `block in run'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:176:in `standard_exception_handling'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:75:in `run'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/bin/rake:33:in `<top (required)>'
/root/.rbenv/versions/2.3.3/bin/rake:22:in `load'
/root/.rbenv/versions/2.3.3/bin/rake:22:in `<main>'
Couldn't create database for {"host"=>"localhost", "encoding"=>"utf8", "adapter"=>"postgis", "username"=>"root", "password"=>"root", "template"=>"template_postgis", "database"=>"inaturalist_development"}
PG::InvalidCatalogName: ERROR:  template database "template_postgis" does not exist
: CREATE DATABASE "inaturalist_test" ENCODING = 'utf8' TEMPLATE = "template_postgis"
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql/database_statements.rb:155:in `async_exec'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql/database_statements.rb:155:in `block in execute'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract_adapter.rb:472:in `block in log'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-4.2.6/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/abstract_adapter.rb:466:in `log'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql/database_statements.rb:154:in `execute'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/connection_adapters/postgresql/schema_statements.rb:78:in `create_database'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/bundler/gems/activerecord-postgis-adapter-4d7c4c91720e/lib/active_record/connection_adapters/postgis_adapter/postgis_database_tasks.rb:24:in `create'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:93:in `create'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:107:in `block in create_current'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:275:in `block in each_current_configuration'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:274:in `each'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:274:in `each_current_configuration'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/tasks/database_tasks.rb:106:in `create_current'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-4.2.6/lib/active_record/railties/databases.rake:17:in `block (2 levels) in <top (required)>'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:240:in `block in execute'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:235:in `each'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:235:in `execute'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:172:in `invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:201:in `block in invoke_prerequisites'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:199:in `each'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:199:in `invoke_prerequisites'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:178:in `block in invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:172:in `invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:201:in `block in invoke_prerequisites'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:199:in `each'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:199:in `invoke_prerequisites'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:178:in `block in invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:172:in `invoke_with_call_chain'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/task.rb:165:in `invoke'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:150:in `invoke_task'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:106:in `each'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:106:in `block in top_level'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:115:in `run_with_threads'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:100:in `top_level'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:78:in `block in run'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:176:in `standard_exception_handling'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/application.rb:75:in `run'
/root/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rake-10.5.0/bin/rake:33:in `<top (required)>'
/root/.rbenv/versions/2.3.3/bin/rake:22:in `load'
/root/.rbenv/versions/2.3.3/bin/rake:22:in `<main>'
Couldn't create database for {"host"=>"localhost", "encoding"=>"utf8", "adapter"=>"postgis", "username"=>"root", "password"=>"root", "template"=>"template_postgis", "database"=>"inaturalist_test"}
psql:/root/inaturalist/db/structure.sql:56: ERROR:  relation "schema_migrations" already exists
psql:/root/inaturalist/db/structure.sql:63: ERROR:  relation "unique_schema_migrations" already exists
psql: FATAL:  database "inaturalist_test" does not exist
rake aborted!
failed to execute:
psql -q -f /root/inaturalist/db/structure.sql inaturalist_test

Please check the output above for any errors and make sure that `psql` is installed in your PATH and has proper permissions.

**Comment by kueda** [comment]
It doesn't seem like you have the `template_postgis` database. Please follow the directions at https://github.com/inaturalist/inaturalist/wiki/development-setup-guide. That will make sure you have all the requirements.

**Comment by BlueSpider67** [comment]
Hi
I can not start  Elasticsearch.
This is error message "Elasticsearch failed to start"

Thanks

**Comment by kueda** [comment]
I'm afraid that's just something you're going to have to debug yourself, or seek help from the elasticsearch team. I'm fairly sure the instructions in the Development Setup Guide work for our dev team, but we all use OS X with pretty similar setups, so if you have something different or are trying to install elasticsearch differently, you might be hitting different issues. If you got that far in our instructions, though, I'm guessing you were able to get the database up and running, so I'm going to close this issue.
